api/provider/aws/test_aws/load_balancer_delete_test.go

```diff
@@ -106,8 +118,96 @@ func TestLoadBalancerDeleteIdempotence(t *testing.T) {
 		DescribeSecurityGroups(gomock.Any()).
 		Return(&ec2.DescribeSecurityGroupsOutput{}, nil)
 
+	mockAWS.EC2.EXPECT().
+		DescribeSecurityGroups(gomock.Any()).
+		Return(&ec2.DescribeSecurityGroupsOutput{}, nil)
+
+	target := provider.NewLoadBalancerProvider(mockAWS.Client(), tagStore, mockConfig)
+	if err := target.Delete("lb_id"); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestLoadBalancerDeleteRetry(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "lb_id",
+			EntityType: "load_balancer",
+			Key:        "name",
+			Value:      "lb_name",
+		},
+		{
+			EntityID:   "lb_id",
+			EntityType: "load_balancer",
+			Key:        "environment_id",
+			Value:      "env_id",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	deleteLBInput := &elb.DeleteLoadBalancerInput{}
+	deleteLBInput.SetLoadBalancerName("l0-test-lb_id")
+
+	mockAWS.ELB.EXPECT().
+		DeleteLoadBalancer(deleteLBInput).
+		Return(&elb.DeleteLoadBalancerOutput{}, nil)
+
+	mockAWS.ELB.EXPECT().
+		DescribeLoadBalancers(gomock.Any()).
+		Return(&elb.DescribeLoadBalancersOutput{}, nil)
+
+	deleteRolePolicyInput := &iam.DeleteRolePolicyInput{}
```

Comment:
We don't care about inputs in this test; just making one comment here even though it applies elsewhere in this test.